### ISSUE TITLE
BspwmDesktopPoolWidget: Improve focus on other screen's desktops

### ISSUE THREAD

**Issue by aruhier** [aruhier/barython#12]
Always swapping is not very effective when wanting to focus a desktop that is attached to another screen because it causes a redraw of the window if the resolution of the 2 screens is not equal.

Prefer swapping only if the target desktop is already focused on another screen, otherwise use "send desktop to monitor" and then focus the targeted desktop. It's particularly convenient with the principle of pool because, that way, the number of desktops attached to a screen will vary, instead of being fixed (as right now).

tldr; Screens have to be able to grab a desktop, so that switching desktops already grabbed by a screen will be faster than always using "swap".
